### PR TITLE
now the cube algorithm counts all the possible combinations between the pairs

### PR DESCRIPTION
**Pull request by giovannifacchinetti99**
Added the list "combination" in the cube algorithm function. In case of >2 particles in the same cube, it computes the combination between all the particles. 
At the end I ran another time, and because of the noise, results are different from before.

### CODE REVIEW

File: doc/utilities/cubes.ipynb [hunk 1]
```diff
@@ -167,7 +168,8 @@
     "        a list containing lists of satelites idx occupying the same cube\n",
     "    \"\"\"\n",
     "    # init\n",
-    "    retval = []\n",
+    "    collision = []\n",
+    "    combination = []\n",
```
Comment (darioizzo) [inline doc/utilities/cubes.ipynb:172]: I would call this variable `pairs`.

File: doc/utilities/cubes.ipynb [hunk 2]
```diff
@@ -167,7 +168,8 @@
     "        a list containing lists of satelites idx occupying the same cube\n",
     "    \"\"\"\n",
     "    # init\n",
-    "    retval = []\n",
+    "    collision = []\n",
```
Comment (darioizzo) [inline doc/utilities/cubes.ipynb:171]: I would call this variable `in_same_cube`

File: doc/utilities/cubes.ipynb (hunk 1 above)
Comment (giovannifacchinetti99) [reply doc/utilities/cubes.ipynb]: Done

File: doc/utilities/cubes.ipynb (hunk 2 above)
Comment (giovannifacchinetti99) [reply doc/utilities/cubes.ipynb]: Done

**Pull request merged**